core/model/generation.go

```diff
@@ -57,6 +58,19 @@ type GenerationApplication struct {
 	ConfigChanges map[string]interface{} `yaml:"config"`
 }
 
+// GenerationCommitApplication represents committed changes to an application
+// made under a generation.
+type GenerationCommitApplication struct {
```

Comment:
Just use `GenerationApplication` for this.
- Add "omitempty" to `UnitProgress`
- `UnitDetail` can hold the same data as `UnitsTracking`. `UnitsPending` will always be empty.